nutils/function.py

```diff
@@ -3052,24 +3053,17 @@ def wrapper( *args, **kwargs ):
         ba.arguments[param.name] = param.default
 
     # replace argument `arg_key` with a derivative helper
-    if isinstance(arg_key, int):
-      args = list(ba.args)
-      orig = args[arg_key]
-      var = DerivativeHelper( orig.shape, arg_axes )
-      args[arg_key] = var
-      kwargs = ba.kwargs
-    elif arg_key in ba.kwargs:
-      args = ba.args
-      kwargs = dict(ba.kwargs)
-      orig = ba.kwargs[arg_key]
-      var = DerivativeHelper( orig.shape, arg_axes )
-      kwargs[arg_key] = var
-    else:
-      raise ValueError( 'argument not found: {!r}'.format( arg_key ) )
+    args = list(ba.args)
+    kwargs = dict(ba.kwargs)
+    keyargs = args if isinstance(arg_key,int) else kwargs
+    orig = keyargs[arg_key]
+    orig_axes = arg_axes if arg_axes is not None else tuple(range(orig.ndim))
```

Comment:
I didn't try but I don't think that's possible; redefining arg_axes would make it local to wrapper so that would make the original argument unreachable.